yt/data_objects/selection_data_containers.py

```diff
@@ -283,16 +283,16 @@ def _generate_container_field_sph(self, field):
             raise KeyError(field)
 
         length = unorm(self.vec)
-        pos = self[self.ds._sph_ptype, "particle_position"]
```

Comment:
How we we know this will always be "_sph_ptypes[0]", why can't it be another index?